File: Lesson-2/Pen.cs
namespace CursoEmVideo_lessons.Lesson_2;

public class Pen
{
    private static string model, color;
    private static int charge;
    private static double point;
    private static bool covered;

    public static int Doodle()
    {
        if (covered == false)
        {
            Console.WriteLine("Doodle");
            return charge--;
        }
        else
        {
            Console.WriteLine("Error! You must uncover first!");
            return charge;
        }
    }

    public static string modelSetter(string var) => model = var;
    public static string colorSetter(string var) => color = var;
    public static int chargeSetter(int var) => charge = var;
    public static double pointSetter(double var) => point = var;
    public static bool coveredSetter(bool var) => covered = var;
    public static bool Cover() => covered = true;
    public static bool Uncover() => covered = false;
}

File: Lesson-2/Pen.ts
namespace CursoEmVideo_lessons.Lesson_2 {
    class Pen {
        model: string; 
        color: string;
        charge: number;
        point: number;
        covered : boolean;

        constructor (
            model: string,
            color: string,
            charge: number,
            point: number,
            covered : boolean
        ){
            this.model;
            this.color;
            this.charge;
            this.point;
            this.covered;
        }

        public Doodle(covered : boolean, charge: number){
            if (covered == false) {
                console.log("Doodle");
                return charge--;
            } else {
                console.log("Error! You must uncover first!");
                return charge;
            }
        }

        public modelSetter(pen: string) {
            this.model = pen;
        }
        public colorSetter(pen: string) {
            this.color = pen;
        }
        public chargeSetter(pen: number) {
            this.charge = pen;
        }
        public pointSetter(pen: number) {
            this.point = pen;
        }
        public coveredSetter(pen: boolean) {
            this.covered = pen;
        }
        public Cover() {
            this.covered = true;
        }
        public Uncover() {
            this.covered = false;
        }
    }
}